gc/gc-base/src/main/java/org/projectnessie/gc/contents/LiveContentSet.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.contents;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import org.immutables.value.Value;
+import org.projectnessie.gc.contents.spi.PersistenceSpi;
+import org.projectnessie.gc.files.DeleteResult;
+import org.projectnessie.gc.files.DeleteSummary;
+import org.projectnessie.gc.files.FileDeleter;
+import org.projectnessie.gc.files.FileReference;
+import org.projectnessie.gc.identify.IdentifyLiveContents;
+
+/**
+ * Represents a set of identified live contents from an {@link
+ * IdentifyLiveContents#identifyLiveContents() identify-live-contents run}.
+ */
+@Value.Immutable
+public abstract class LiveContentSet {
+
+  public static ImmutableLiveContentSet.Builder builder() {
+    return ImmutableLiveContentSet.builder();
+  }
+
+  public ImmutableLiveContentSet.Builder unbuild() {
+    return ImmutableLiveContentSet.builder().from(this);
+  }
+
+  abstract PersistenceSpi persistenceSpi();
+
+  public abstract UUID id();
+
+  public abstract Instant created();
+
+  @Nullable
+  public abstract Instant identifyCompleted();
+
+  @Nullable
+  public abstract Instant expiryStarted();
+
+  @Nullable
+  public abstract Instant expiryCompleted();
+
+  public abstract Status status();
+
+  @Nullable
+  public abstract String errorMessage();
+
+  public void delete() {
```

Comment:
I don't see caller for this. We should call this when expiry() is success?

Reply:
See markdown in #4991